pkg/resourcecollector/resourcecollector.go

```diff
@@ -522,6 +523,37 @@ func (r *ResourceCollector) GetResources(
 	return allObjects, pvcsWithOwnerReference, nil
 }
 
+// gatherResourceInChunks() collects all the resources present per ns or resource types.
+// It does it in batch of 500 unless specified by user via a config-map. This function
+// primarily helps in avoiding server timeout error when the number of resource or size
+// of it large.
+func gatherResourceInChunks(dynamicClient dynamic.ResourceInterface, opts Options, selectors string) (*unstructured.UnstructuredList, error) {
+	var err error
+	fn := "gatherResourceInChunks"
+	objectsList := &unstructured.UnstructuredList{}
+	var temp *unstructured.UnstructuredList
+	listOps := metav1.ListOptions{
+		LabelSelector: selectors,
+		Limit:         opts.ResourceCountLimit,
+	}
+	for {
+		temp, err = dynamicClient.List(context.TODO(), listOps)
+		if err != nil {
+			logrus.Infof("%v: failed to list resources", fn)
+			return nil, err
+		}
+		if len(temp.GetContinue()) != 0 {
```

Comment:
There could be a potential problem, when we are fetching the last chunk as i understand GetContinue() would be zero as there is nothing more to fetch, in such a case last chunk is missed. Is my understanding correct?

Reply:
yes  I just figured that out while doing unit testing .. hence added that check now..

Reply:
By looking at the code I think, it's already handled and we don't need an additional append. As `temp.GetContinue()` is called before the append. So for the last chunk, the `temp.GetContinue()` will be zero and its appended to the list items and on the next iteration it just avoids the if clause

Reply:
Sorry for the noise, i just misunderstood :)